build.sbt

```diff
@@ -8,18 +8,44 @@ inThisBuild(Seq(
    scalaVersion := "2.12.8",
    crossScalaVersions := Seq("2.12.8", "2.11.8"),
    scalacOptions ++= Seq(
-      "-unchecked",
-      "-deprecation",
-      "-feature",
-      "-language:higherKinds",
-      "-language:postfixOps"
+      "-unchecked",                        // Show details of unchecked warnings.
+      "-deprecation",                      // Show details of deprecation warnings.
+      "-encoding", "UTF-8",                // Set correct encoding for Scaladoc.
+      "-feature",                          // Show details of feature warnings.
+      "-explaintypes",                     // Explain type errors in more detail.
+      "-Xcheckinit",                       // Wrap field accessors to throw an exception on uninitialized access.
+      "-Xlint",                            // Ensure best practices are being followed.
+      "-Xlint:adapted-args",               // Warn if an argument list is modified to match the receiver.
+      "-Xlint:constant",                   // Evaluation of a constant arithmetic expression results in an error.
```

Comment:
Some of these flags are not supported in 2.11:
```
-Xlint:constant
-Ywarn-extra-implicit
-Ywarn-unused:*
```
We might need either to drop 2.11 support, or to add a version switch.

Reply:
Version switch: https://github.com/gawkermedia/macrame/pull/8

Reply:
We'll deal with the version number before releasing 2.13